inbm/integration-reloaded/scripts/create_signature.py

```diff
@@ -18,12 +18,12 @@
     if num_params == 4:
         password = sys.argv[3].encode('utf-8')
 
-with open(package_name) as package:
-    checksum = hashlib.sha384(package.read().encode('utf-8')).hexdigest()
+with open(package_name, 'rb') as package:
+    checksum = hashlib.sha384(package.read()).hexdigest()
 
-with open(file_name) as f:
-    priv_key = load_pem_private_key(f.read().encode('utf-8'), password=password, backend=default_backend())
+with open(file_name, 'rb') as f:
+    priv_key = load_pem_private_key(f.read(), password=password, backend=default_backend())
 
 signature = priv_key.sign(checksum.encode('utf-8'), padding.PSS(mgf=padding.MGF1(hashes.SHA384()), salt_length=padding.PSS.MAX_LENGTH), hashes.SHA384())
 
-print((hexlify(signature)).decode('utf-8', errors='strict'))
\ No newline at end of file
+print((hexlify(signature)).decode('utf-8', errors='strict'))
```

Comment:
**Suggestion:** Use a context manager to ensure that the signature is always printed, even if an exception occurs during the signing process. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
try:
    signature = priv_key.sign(checksum.encode('utf-8'), padding.PSS(mgf=padding.MGF1(hashes.SHA384()), salt_length=padding.PSS.MAX_LENGTH), hashes.SHA384())
finally:
    print((hexlify(signature)).decode('utf-8', errors='strict'))
```

Reply:
we definitely don't want to print the signature if there's an exception.. it would just be None at best